File: aoc-2023/src/test/java/com/capital7software/aoc/aoc2023/days/Day09Test.java
package com.capital7software.aoc.aoc2023.days;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

class Day09Test extends AdventOfCodeTestBase {
  private static final Logger LOGGER = LoggerFactory.getLogger(Day09Test.class);

  @BeforeEach
  void setUp() {
    var instance = new Day09();
    setupFromFile(instance.getDefaultInputFilename());
  }

  @Test
  void sumOfNextPredictedHistories() {
    var instance = new Day09();
    int expected = 114;
    var actual = instance.sumOfNextPredictedHistories(lines);
    assertEquals(expected, actual, "The total of next predicted values "
        + "is not what was expected: " + expected);
  }

  @Test
  void sumOfPreviousPredictedHistories() {
    var instance = new Day09();
    int expected = 2;
    var actual = instance.sumOfPreviousPredictedHistories(lines);
    assertEquals(expected, actual, "The total of previous predicted values "
        + "is not what was expected: " + expected);
  }

  @Override
  protected Logger getLogger() {
    return LOGGER;
  }
}

File: aoc-2017/src/test/kotlin/com/capital7software/aoc/aoc2017aoc/days/Day09Test.kt
package com.capital7software.aoc.aoc2017aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day09Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day09Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day09()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testScoreOfStream() {
    val instance = Day09()
    val expected = listOf(1, 6, 5, 16, 1, 9, 9, 3)

    lines.withIndex().forEach { (index, line) ->
      val actual = instance.scoreOfStream(line)

      Assertions.assertEquals(
          expected[index], actual, "The largest value " +
          "is not what was expected: $expected"
      )
    }
  }

  @Test
  fun testGarbageRemoved() {
    setupFromFile("inputs/input_day_09-02.txt")
    val instance = Day09()
    val expected = listOf(0, 4, 3, 2, 0, 0, 10)

    lines.withIndex().forEach { (index, line) ->
      val actual = instance.garbageRemoved(line)

      Assertions.assertEquals(
          expected[index], actual, "The largest value " +
          "is not what was expected: $expected"
      )
    }
  }
}
